src/hdmf/common/hierarchicaltable.py

```diff
@@ -0,0 +1,258 @@
+                index_data = ([dynamic_table.id[row_index], ] +
+                              [dynamic_table[row_index, colname]
+                               for colname in dynamic_table.colnames if colname != hcol_name])
+                # for i in indexed_column_indicies:  # Fix data from indexed columns
+                #    index_data[i] = tuple(index_data[i])  # Convert from list to tuple (which is hashable)
+                index.append(tuple(index_data))
+                # Determine the names for our index and columns of our output table if this is the first row.
+                # These are constant for all rows so we only need to do this onle once for the first row.
+                if row_index == 0:
+                    index_names = ([(dynamic_table.name, 'id')] +
+                                   [(dynamic_table.name, colname)
+                                    for colname in dynamic_table.colnames if colname != hcol_name])
+                    columns = pd.MultiIndex.from_tuples([(hcol_target.name, 'id'), ] +
+                                                        [(hcol_target.name, c) for c in row_df.columns],
+                                                        names=('source_table', 'label'))
+        #  if we had an empty data table then at least define the columns
+        if index_names is None:
+            index_names = ([(dynamic_table.name, 'id')] +
+                           [(dynamic_table.name, colname)
+                            for colname in dynamic_table.colnames if colname != hcol_name])
+            columns = pd.MultiIndex.from_tuples([(hcol_target.name, 'id'), ] +
+                                                [(hcol_target.name, c) for c in hcol_target.colnames],
+                                                names=('source_table', 'label'))
+
+    # Case 2:  Our DynamicTableRegion columns points to another table with a DynamicTableRegion
+    else:
+        # 1) First we need to recursively flatten the hierarchy by calling 'to_hierarchical_dataframe()'
+        #    (i.e., this function) on the target of our hierarchical column
+        #    First we here get a list of DataFrames, one for each row. If our hcol is a VectorIndex
+        #    (i.e., out column is a ragged array), then simply loading the data from the VectorIndex
+        #    will do the trick. If we have a regular VectorData column, then we need to load the
+        #    elements ourselves (using slice syntax to make sure we get DataFrames) one-row-at-a-time
+        hcol_hdf = to_hierarchical_dataframe(hcol_target)
+        if isinstance(hcol, VectorIndex):
+            rows = hcol.get(slice(None), index=False, df=True)
+        else:
+            rows = [hcol[i:(i+1)] for i in range(len(hcol))]
+        # 2) Iterate over all rows, where each row is described by a DataFrame with one-or-more rows
+        for row_index, row_df_level1 in enumerate(rows):   # need hcol[:] here  in case this is an h5py.Dataset
```

Comment:
What does the comment on this line refer to?

Reply:
```suggestion
        for row_index, row_df_level1 in enumerate(rows):
```